pkg/testing/fixture.go

```diff
@@ -1209,7 +1209,7 @@ func createTempDir(t *testing.T) string {
 
 	cleanup := func() {
 		if !t.Failed() {
-			if err := os.RemoveAll(tempDir); err != nil {
+			if err := removeAll(tempDir); err != nil {
```

Comment:
We actually already have this logic done and used by Elastic Agent. Easier to just use that instead of duplicating the code.

https://github.com/elastic/elastic-agent/blob/main/internal/pkg/agent/install/uninstall.go#L164